### DISCUSSION TITLE
UEFI libraries for C++

### DISCUSSION THREAD

**Discussion by ghost** [community/community#114780] [Programming Help]
### Body

Hi,

I am making an MS DOS remake in C++ and possibly Assembly language for optimization. The OS will work on the original 16-bit environment, but also supports 32-bit mode. Can you recommend a C++ library to ease development of the OS?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ethanwater** [comment]
QEMU is a good one.
